### PR TITLE
support building on MinGW part 1

### PR DESCRIPTION
**Pull request by franz**

### CODE REVIEW

File: lib/CL/pocl_util.c [hunk 1]
```diff
@@ -24,12 +24,24 @@
...
+
 #include <errno.h>
 #include <limits.h>
 #include <stdlib.h>
 #include <string.h>
 #include <ctype.h>
 
+#if defined(__FreeBSD__)
+#include <stdlib.h>
+#elif defined(__WIN32)
```
Comment (linehill) [inline lib/CL/pocl_util.c:39]: I think, MSVC does not define `__win32`.
```suggestion
#elif defined(_WIN32)
```

File: lib/CL/pocl_threads.cc [hunk 2]
```diff
@@ -0,0 +1,206 @@
+/* OpenCL runtime library: utility functions for thread operations
```
Comment (pjaaskel) [inline lib/CL/pocl_threads.cc:1]: ...for C++-supported hosts.

File: lib/CL/pocl_threads_c.h [hunk 3]
```diff
@@ -0,0 +1,154 @@
+/* OpenCL runtime library: utility functions for thread operations
```
Comment (pjaaskel) [inline lib/CL/pocl_threads_c.h:1]: Implemented using pthreads. Maybe the file name should reflect it as well. It's not using the C standard but the POSIX threads standard, if we are strict here. pocl_threads_posix.h?

File: lib/CL/pocl_timing.cc [hunk 4]
```diff
@@ -0,0 +1,56 @@
+/* OpenCL runtime library: time measurement utility functions
```
Comment (pjaaskel) [inline lib/CL/pocl_timing.cc:1]: ...implemented using C++11 (?).

**Review by pjaaskel — APPROVED** [review]
Good job. Just a couple of insights you can fix post pull. I suggest @linehill to quickly peruse this before we pull it in.

**Pull request merged**